book/src/pil/declarations.md

```diff
@@ -7,18 +7,21 @@ from the type of the symbol and the way the symbol is used.
 Symbols can be declared using ``let <name>;`` and they can be declared and defined
 using ``let <name> = <value>;``, where ``<value>`` is an expression. The [type](./types.md) of the symbol
 can be explicitly specified using ``let <name>: <type>;`` and ``let <name>: <type> = <value>;``.
+It is even possible to define generic symbols using ``let<TV1, TV1, ..> <name>: <type> = <value>;``,
+where the `TV` are newly created type variables that can be used in the type.
 
 This syntax can be used for constants, fixed columns, witness columns and even (higher-order)
 functions that can transform expressions. The kind of symbol is deduced by its type and the
 way the symbol is used:
 
-- Symbols without a value are witness columns or arrays of witness columns. Their type can be omitted. If it is given, it must be ``col`` or ``col[]``.
-- Symbols evaluating to a number or with type ``fe`` are constants.
-- Symbols without type but with a value that is a function with a single parameter are fixed columns.
-- Symbols defined with a value and type ``col`` (or ``col[]``) are fixed columns (or arrays of fixed columns).
+- Symbols without a value are witness columns or arrays of witness columns. Their type can be omitted. If it is given, it must be ``col`` or ``col[k]``.
+- Symbols with type ``fe`` are constants.
+- Symbols defined with a value and type ``col`` (or ``col[k]``) are fixed columns (or arrays of fixed columns).
+- Symbols defined with a value and type ``expr`` (or ``expr[k]``) are intermediate columns (or arrays of intermediate columns).
 - Everything else is a "generic symbol" that is not a column or constant.
```

Comment:
I don't get the distinction between a generic symbol and a constant. Isn't every definition constant?